### PR TITLE
support truncate postgres command

### PR DESCRIPTION
**Pull request by stlava**
The goal of this PR is to handle `truncate table ...` commands. Currently bifrost's underlying parser (`parselogical`) is unable to parse these commands and throws a parsing error which causes bifrost to shutdown for safety.

### CODE REVIEW

File: parselogical/parselogical.go [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+	Operation   string                 // filled if this is a DML statement with the name of the operation (INSERT/UPDATE/DELETE)
+	NoTupleData bool                   // true if this had no actual tuple data
+	Columns     map[string]ColumnValue // if a DML statement, the fields affected by the operation with their values and types
+	OldColumns  map[string]ColumnValue // if an UPDATE and REPLICA IDENTITY setting (FULL or USING INDEX both will add values here)
+}
+
+// NewParseResult creates the structure that is filled by the Parse operation
+func NewParseResult(msg string) *ParseResult {
+	pr := new(ParseResult)
+	fmt.Println(msg)
```
Comment (ruban-rengaraju) [inline parselogical/parselogical.go:75]: print leftover from testing?

Comment (stlava) [reply parselogical/parselogical.go:75]: Ah yup. Thanks for catching!

File: parselogical/parselogical.go [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+				} else {
+					prev := state.Prev
+					state.Prev = state.Current
+					state.Current = prev
+				}
+			}
+		}
+	}
+
+	if state.Current == parseStateOperationTruncate {
```
Comment (ruban-rengaraju) [inline parselogical/parselogical.go:280]: it looks like the main change to the parser is to allow for messages with 'TRUNCATE' operations to be passed downstream to the rest of the pg-bifrost handlers. Presumably you've been able to run & test this end-to-end locally?

Comment (ruban-rengaraju) [reply parselogical/parselogical.go:280]: oh nice, i see the additional itests as well.

**Review by ruban-rengaraju — APPROVED** [review]

File: parselogical/parselogical.go (hunk 2 above)
Comment (stlava) [reply parselogical/parselogical.go:280]: Yeah I added an integration test.

**Pull request merged**